10-ncvs-vignette.Rmd

```diff
@@ -0,0 +1,599 @@
+
+```{r}
+#| label: inc-sum
+
+inc_hh_sums <-
+  inc_ind %>%
+  filter(V4529_num > 23) %>% # restrict to household crimes
+  group_by(YEARQ, IDHH) %>%
+  summarize(
+    WGTVICCY = WGTVICCY[1],
+    across(starts_with("Property"), ~ sum(. * serieswgt),
+           .names = "{.col}"),
+    .groups = "drop"
+  )
+
+inc_pers_sums <-
+  inc_ind %>%
+  filter(V4529_num <= 23) %>% # restrict to person crimes
+  group_by(YEARQ, IDHH, IDPER) %>%
+  summarize(WGTVICCY = WGTVICCY[1],
+            across(c(
+              starts_with("Violent"), starts_with("AAST")
+            ),
+            ~ sum(. * serieswgt), .names = "{.col}"),
+            .groups = "drop")
+```
+
+The victimization summary files are then merged onto the appropriate files. For any record on the household or person file that is not on the victimization file, the victimization counts are set to 0 after merging. In this step, we will also create the victimization adjustment factor. See 2.2.4 in the User's Guide for details of why this adjustment is created. It is calculated as follows:
+
+$$ A_{ijk}=\frac{v_{ijk}}{w_{ijk}}$$ 
+
+where $w_{ijk}$ is the person weight (`WGTPERCY`) for personal crimes or the household weight (`WGTHHCY`) for household crimes and $v_{ijk}$ is the victimization weight (`WGTVICCY`) for household $i$, respondent $j$, in reporting period $k$. The adjustment factor is set to 0 if no incidents are reported.
+
+```{r}
+#| label: merge-inc-sum
+
+hh_in <-
+  read_rds(here::here("AnalysisData", "ncvs_2021_household.rds"))
+
+pers_in <-
+  read_rds(here::here("AnalysisData", "ncvs_2021_person.rds"))
+
+# Set up a list of 0s for each crime type/characteristic to replace NA's
+hh_z_list <- rep(0, ncol(inc_hh_sums) - 3) %>% as.list() %>%
+  setNames(names(inc_hh_sums)[-(1:3)])
+pers_z_list <- rep(0, ncol(inc_pers_sums) - 4) %>% as.list() %>%
+  setNames(names(inc_pers_sums)[-(1:4)])
+
+hh_vsum <- hh_in %>%
+  full_join(inc_hh_sums, by = c("YEARQ", "IDHH")) %>%
+  replace_na(hh_z_list) %>%
+  mutate(ADJINC_WT = if_else(is.na(WGTVICCY), 0, WGTVICCY / WGTHHCY))
+
+pers_vsum <- pers_in %>%
+  full_join(inc_pers_sums, by = c("YEARQ", "IDHH", "IDPER")) %>%
+  replace_na(pers_z_list) %>%
+  mutate(ADJINC_WT = if_else(is.na(WGTVICCY), 0, WGTVICCY / WGTPERCY))
+```
+
+### Derived demographic variables
+
+A final step in file preparation for the household and person files is creating any derived variables on the household and person files, such as income categories or age categories, for subgroup analysis. This step can be done before or after merging on the victimization counts.
+
+For the household file, we create categories for tenure (rental status), urbanicity, income, place size, and region. For the person file, we create categories for sex, race/Hispanic origin, age categories, and marital status. We also merge on the household demographics to the person file as well as the design variables (`V2117` and `V2118`).
+
+```{r}
```

Comment:
Maybe we can sketch out what is happening with each of these files - the code is easy to follow, but there is quite a bit!